extras/wait-for-bricks.sh

```diff
@@ -0,0 +1,159 @@
+                fi
+
+                if [ -z $number_of_bricks ];
+                then
+                        echo 0
+                else
+                        echo $number_of_bricks
+                fi
+        done | tail -1
+}
+
+function check_brick_status() {
+        local volname=$1
+        local daemon=$2
+
+        cmd="gluster --xml volume status $volname"
+        if [[ -z $daemon ]]
+         then
+                 echo `$cmd | grep '<status>1' | wc -l`
+        else
+                 echo `$cmd | sed -n -e "/${daemon}/,/<status>/ p" | grep '<status>1' | wc -l`
+         fi
+}
+
+function volume_online_brick_count()
+{
+        local volname=$1
+        local v1=0
+        local v2=0
+        local v3=0
+        local v4=0
+        local v5=0
+        local tot=0
+
+        #First count total Number of bricks and then subtract daemon status
+        v1=`check_brick_status $volname`
+        v2=`check_brick_status $volname "Self-heal"`
+        v3=`check_brick_status $volname "Quota"`
+        v4=`check_brick_status $volname "Snapshot"`
+        v5=`check_brick_status $volname "Tier"`
+        tot=$((v1-v2-v3-v4-v5))
+        echo $tot
+}
+
+function online_brick_count()
+{
+        if [ "0" == $(volinfo_field "all" "count") ];
+        then
+                echo 0
+                return 0
+        fi
+        gluster volume list | while read volname;
+        do
+                if [ "Started" == $(volinfo_field $volname 'statusStr') ];
+                then
+                        if [ "Y" == $(is_enabled $(volinfo_option $volname features.shard)) ];
+                        then
+                                brick_count=$(volume_online_brick_count $volname)
+                                number_of_bricks=$((number_of_bricks + brick_count))
+                        fi
+                fi
+
+                if [ -z $number_of_bricks ];
+                then
+                        echo 0
+                else
+                        echo $number_of_bricks
+                fi
+        done | tail -1
+}
+
+function check_glusterd_running()
+{
+        pidof glusterd 2>&1 > /dev/null
+        if [ "$?" -ne 0 ]; then
+                echo "Glusterd is not running";
+                exit 1;
+        fi
+}
+
+function wait_for_bricks_up()
+{
+        local endtime=$1
+
+        while [ $(expected_online_brick_count) -gt $(online_brick_count) ]; do
+                local now=$(date +%s%N)
+                if [[ $now -ge $endtime ]]; then
+                        echo "Timeout Expired, online/expected-online brick-counts: $(online_brick_count)/$(expected_online_brick_count)"
+                        exit 1
+                fi
+                sleep 1;
+        done
+}
+
+check_glusterd_running;
+timeout=$1
+start=$(date +%s%N)
+endtime=$(( ${timeout}*1000000000 +start ))
+wait_for_bricks_up $endtime
+now=$(date +%s%N)
+echo "All bricks for Block Hosting Volumes came online in $(((now-start)/1000000000)) seconds"
+exit 0;
```

Comment:
Why we had to return success at the end ? It make-sense to fail in case bricks won't come by timeout IMO.

Reply:
IMO, return success at the end here will be okay. And in the case bricks won't come by timeout just print out the error or warning logs should be okay.

Because there maybe some commands executing errors in the script, and the success at the end means there has no errors for the script itself.

Reply:
Oh! Looks like, we are 'exit 1' at line 146, I think this should be fine then. Sorry my fault.